lib/rspec/instafail.rb

```diff
@@ -4,14 +4,14 @@ module RSpec
   lib = File.expand_path(File.dirname(File.dirname(__FILE__)))
   $LOAD_PATH << lib unless $LOAD_PATH.include?(lib)
 
-  if defined?(::RSpec::Core) && ::RSpec::Core::Version::STRING >= '3.0.0'
-    require "rspec/instafail/rspec_3"
-  else
-    begin
+  if defined?(::RSpec::Core)
+    if ::RSpec::Core::Version::STRING >= '3.0.0'
+      require "rspec/instafail/rspec_3"
+    else
       require "rspec/instafail/rspec_2"
-    rescue LoadError # try rspec 1
-      require "rspec/instafail/rspec_1"
     end
+  elsif defined?(::Spec::VERSION)
```

Comment:
If you do that then I belive you'll get an error when running rake tasks when using rspec 1.x, as it's the same issue.  Rspec isn't loaded and instalfail with raise an exception expecting rspec to be loaded.

We could try to require 'rspec/core' at the top, if it works, then test for the v3 version, catch the exception if 'rspec/core' doesn't load and then require 'spec' and load v1.

what do you think?